src/test/java/com/parasoft/findings/utils/common/util/IOUtilsTest.java

```diff
@@ -49,22 +87,56 @@ public void testCloseReader_withIOException() throws IOException {
         assertThrows(IOException.class, reader::close);
     }
 
+    /**
+     * Test {@link IOUtils#close(Writer)}
+     */
     @Test
-    public void testCloseWriter_normal() throws IOException {
+    public void testClose4_normal() throws IOException {
         Writer writer = mock(Writer.class);
 
         IOUtils.close(writer);
 
         verify(writer, Mockito.atLeastOnce()).close();
     }
 
+    /**
+     * Test {@link IOUtils#close(Writer)}
+     */
     @Test
-    public void testCloseWriter_withIOException() throws IOException {
+    public void testClose4_withIOException() throws IOException {
         Writer writer = mock(Writer.class);
         doThrow(ioException).when(writer).close();
 
         IOUtils.close(writer);
 
         assertThrows(IOException.class, writer::close);
     }
+
+    @Test
+    public void testMkdirs() {
+        File dir = new File("src/test/resources/xml/staticanalysis");
+        File preparedFile = new File("src/test/resources/xml/staticanalysis/", "cpptest_pro_report_202001.xml");
+        File wrongFile = new File( "");
+        //When given path is null
+        assertNull(IOUtils.mkdirs(null));
+        //When given path is directory
+        assertEquals(dir, IOUtils.mkdirs(dir.getPath()));
+        //When file is already exists
+        assertNull(IOUtils.mkdirs(preparedFile.getPath()));
+        //When file could not be created
+        assertNull(IOUtils.mkdirs(wrongFile.getPath()));
+    }
+
+    @Test
+    public void testCopy() throws IOException {
+        File sourceFile = new File("src/test/resources/xml/staticanalysis/", "cpptest_pro_report_202001.xml");
+        File destinationFile = new File("src/test/resources/xml/", "destinationFile.xml");
+        InputStream source = Files.newInputStream(sourceFile.toPath());
+        OutputStream destination = Files.newOutputStream(destinationFile.toPath());
+
+        assertEquals(0, destinationFile.length());
+        IOUtils.copy(source, destination);
+        assertEquals(sourceFile.length(), destinationFile.length());
+        destinationFile.delete();
```

Comment:
Use the way bellow to delete newly generated files or dir.
`try {
  ...
} finally{
  if (destinationFile.exists()) {
    destinationFile.delete();
  }
}`

There are other palces need to do the same operation.